src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/SecondarySite/index.js

```diff
@@ -29,154 +29,219 @@ const StepPrimarySetup = () => {
 	};
 
+	const changePrimaryPrev = () => {
+		const idx = primaryTypeList.findIndex( ( val ) => primaryType === val );
+
+		if ( idx === 0 )
+			changePrimaryType( primaryTypeList[ primaryTypeList.length - 1 ] );
+		else changePrimaryType( primaryTypeList[ idx - 1 ] );
+	};
+
+	const changePrimaryNext = () => {
+		const idx = primaryTypeList.findIndex( ( val ) => primaryType === val );
+
+		if ( idx === primaryTypeList.length - 1 )
+			changePrimaryType( primaryTypeList[ 0 ] );
+		else changePrimaryType( primaryTypeList[ idx + 1 ] );
+	};
+
+	const secondarySiteTypeChips = () => {
+		let idx = 0;
+		const secondaryChipList = [];
+
+		const types = siteClassData?.types[ primaryType ]?.secondaryTypes;
+		for ( const typeKey in types ) {
+			secondaryChipList.push(
+				<div
+					key={ types[ typeKey ]?.slug }
+					role="button"
+					tabIndex={ idx + 1 }
+					className={ `${
+						types[ typeKey ].slug === secondaryType
+							? 'chosenSecondaryCategory '
+							: ''
+					}nfd-card-sec-category` }
+					onClick={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+					onKeyDown={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+				>
+					<span className="categName">
+						{ types[ typeKey ]?.label }
+					</span>
+				</div>
+			);
+			idx++;
+		}
+
+		return secondaryChipList;
+	};
+
 	return (
 		<CommonLayout isBgPrimary isCentered>
 			<NewfoldLargeCard>
 				<div className="nfd-card-heading center">
 					<CardHeader
-						heading={ __(
-							currentStep?.heading,
-							'wp-module-onboarding'
-						) }
-						subHeading={ sprintf(
-							__( content.subHeading, 'wp-module-onboarding' ),
-							translations( 'SITE' )
-						) }
-						question={ __(
-							currentStep?.subheading,
-							'wp-module-onboarding'
-						) }
+						heading={ contents?.cardHeading }
+						subHeading={ contents.subHeading }
+						question={ contents?.question }
 					/>
 				</div>
-				<Animate
-					type="fade-in-disabled"
-					after={
-						categoriesArray[ 0 ]?.subCategories &&
-						selectedCategoryInStore !== null
-					}
-				>
+				<Animate type="fade-in-disabled" after={ siteClassData }>
 					<div className="nfd-setup-secondary-categories">
-						<div className="nfd-card-category-wrapper">
-							<div className="category-scrolling-wrapper">
-								<span
-									className="icon"
-									style={ {
-										backgroundImage:
-											categoriesArray[ 0 ].icon,
-									} }
-								/>
-								<p className="categName">
-									{ ' ' }
-									{ categoriesArray[ 0 ].name }
-								</p>
-							</div>
-						</div>
-
-						<div className="subCategoriesSection">
-							{ categoriesArray[ 0 ]?.subCategories?.map(
-								( item, idx ) => {
-									return (
+						<div className="nfd-card-sec-category-wrapper">
+							{ siteClassData && (
+								<div className="category-scrolling-wrapper">
+									<div className="category-scrolling-wrapper_left-btn">
```

Comment:
```suggestion
									<div className="category-scrolling-wrapper__left-btn">
```
We should use BEM (please fix other occurrences as well): https://getbem.com/naming/